department-of-veterans-affairs/caseflow
# Queue | Add document ID validations during attorney check out

Resolves #5993 

### Description
Adds validation to `document_id` within Attorney checkout flow (SubmitDecisionView)

All OMO decisions should have a `document_id` with one of two formats:
New:
```regex
/^\d{5}-\d{8}$/
```
Old (decisions not prepared using the IDT):
```regex
/^\d{8}\.\d{3,4}$/
```

VHA work product document id format:
```regex
/^V\d/{7}\.\d{3,4}$/
```

IME work product document id format:
```regex
/^M\d{7}\.\d{3,4}$/
```

### Acceptance Criteria 
- [ ] Within SubmitDecisionView, Document ID should validate to either generic regex / specific old work product type regexes depending on OMO type

### Testing Plan
1. Go to `/queue` as attorney (`BVASCASPER1`), case detail view, Medical Request Ready for Review
2. Confirm Document ID validates correctly

<details><summary>gif</summary>
<img src="https://user-images.githubusercontent.com/8533004/42842296-c8e237ba-89da-11e8-916a-eff47dcddc8a.gif">
</details>
